kvserver: improve snapshot sending msg

tbg opened the pull request:
Old:

> streamed snapshot to (n9,s9):7LEARNER: kv pairs: 35321, log entries:
> 0, rate-limit: 8.0 MiB/sec, 2256.83s

New:

> streamed LEARNER snapshot 0e9dea53 at applied index 17 to
> (n3,s3):2LEARNER in 0.00s @ 0 B/s: kv pairs: 10, log entries: 0,
> rate-limit: 8.0 MiB/s, queued: 0.00s

Note how this distinguishes between the time spent queued in the
receiver's snapshot semaphore vs the actual sending, and also
has the actual transfer rate.

cc @miretskiy 

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48579)
<!-- Reviewable:end -->

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48579#-:-M6qc5xi8KQHAqjnA95l:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/kv/kvserver/store_snapshot.go, line 105 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48579#-M6qbJf90iXJNOKw29li:-M6qbJfA52ziFR6z2SP4:b-8ajjki) ([raw file](https://github.com/cockroachdb/cockroach/blob/13ac89e65fb530065bad7fe26f590e8e1c84f0c7/pkg/kv/kvserver/store_snapshot.go#L105)):*
> ```Go
> 	newBatch func() storage.Batch
> 	// bytesSent is updated in sendBatch and returned from Send(). It does not
> 	// reflect the log entries sent, which is fine since that's essentially dead
> ```

what code is dead? I am curious, but also I think this comment might be confusing for most readers.
___
*[pkg/kv/kvserver/store_snapshot.go, line 939 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48579#-M6qbfIx-h0khBMSqrMU:-M6qbfIx-h0khBMSqrMV:bmyuce3) ([raw file](https://github.com/cockroachdb/cockroach/blob/13ac89e65fb530065bad7fe26f590e8e1c84f0c7/pkg/kv/kvserver/store_snapshot.go#L939)):*
> ```Go
> 	}
> 
> 	durQueued := timeutil.Since(start)
> ```

consider adding some words about queuing and about this streaming protocol above. This is the first hint at any queuing.


<!-- Sent from Reviewable.io -->

tbg committed:
kvserver: improve snapshot sending msg

Old:

> streamed snapshot to (n9,s9):7LEARNER: kv pairs: 35321, log entries:
> 0, rate-limit: 8.0 MiB/sec, 2256.83s

New:

> streamed LEARNER snapshot 0e9dea53 at applied index 17 to
> (n3,s3):2LEARNER in 0.00s @ 0 B/s: kv pairs: 10, log entries: 0,
> rate-limit: 8.0 MiB/s, queued: 0.00s

Note how this distinguishes between the time spent queued in the
receiver's snapshot semaphore vs the actual sending, and also
has the actual transfer rate.

Release note: None

tbg reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48579)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei)

___
*[pkg/kv/kvserver/store_snapshot.go, line 105 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48579#-M6qbJf90iXJNOKw29li:-M77D2wH0EjPMA2mMrNx:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/13ac89e65fb530065bad7fe26f590e8e1c84f0c7/pkg/kv/kvserver/store_snapshot.go#L105)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

what code is dead? I am curious, but also I think this comment might be confusing for most readers.
</blockquote></details>

Done.
___
*[pkg/kv/kvserver/store_snapshot.go, line 939 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48579#-M6qbfIx-h0khBMSqrMU:-M77DOiHDoe2xwhMWIcR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/13ac89e65fb530065bad7fe26f590e8e1c84f0c7/pkg/kv/kvserver/store_snapshot.go#L939)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

consider adding some words about queuing and about this streaming protocol above. This is the first hint at any queuing.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

tbg commented:
TFTR!

bors r=andreimatei
